### DISCUSSION TITLE
你好 我想请教一下关于ReactiveCommand.Create()

### DISCUSSION THREAD

**Discussion by luojunyuan** [HMBSbige/BilibiliLiveRecordDownLoader#154] [Q&A]
通过 ReactiveCommand.CreateFromObservable 初始化Command与 ReactiveCommand.Create() 有什么不同吗？

我在 [SettingViewModel.cs](https://github.com/HMBSbige/BilibiliLiveRecordDownLoader/blob/master/BilibiliLiveRecordDownLoader/ViewModels/SettingViewModel.cs) 中看到比如 `SelectDirectory()` 是一个普通的方法，`OpenDirectory()` 是使用Observable.Start(() => Unit.Default) 这样的形式创建的Observable。直接把OpenDirectory() 改为普通方法怎么样

**Comment by HMBSbige** [comment]
应该是一样的

**Reply by luojunyuan** [reply]
谢谢回复，从BilibiliLiveRecordDownLoader学了好多。reactive真是强大😊

**Reply by HMBSbige** [reply]
不过我现在看看已经写得不是很好了，之后有空还得再改改😂

**Comment by luojunyuan** [comment]
https://www.reactiveui.net/reactive-extensions/start/observable.start

这才从上面链接了解到Observable.Start()是异步的，从另一个线程执行操作了。直接ReactiveCommand.Create()就是当前线程一般是ui线程。似乎是这样
